sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClientServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Net.NetworkInformation;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Security.KeyVault.Secrets;
+
+/// <summary>
+/// TODO
+/// </summary>
+public static class SecretClientServiceCollectionExtensions
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    /// <param name="services"></param>
+    /// <returns></returns>
+    public static IServiceCollection AddSecretClient(this IServiceCollection services)
+    {
+        services.AddCommonOptions();
+
+        services.AddOptions<SecretClientOptions>()
+            .Configure<IOptions<ClientPipelineOptions>>((clientOptions, commonOptions) =>
+            {
+                clientOptions.LoggerFactory = commonOptions.Value.Logging.LoggerFactory;
```

Comment:
I think we can get this differently -- an Azure.Core-based client shouldn't need to rely on SCM pipeline options to get ILoggerFactory.